pytorch_lightning/loops/epoch/training_epoch_loop.py

```diff
@@ -161,8 +142,11 @@ def advance(self, dataloader_iter: Iterator, **kwargs: Any) -> None:
         processed_batch_end_outputs = self._prepare_outputs(batch_end_outputs, batch_mode=True)
 
         # hook
-        if not isinstance(self.batch_loop, IteratorBatchProcessor):
-            self.trainer.call_hook("on_train_batch_end", processed_batch_end_outputs, batch, self.batch_idx, 0)
+        self.trainer.call_hook("on_train_batch_end", processed_batch_end_outputs, batch, self.batch_idx, 0)
```

Comment:
It's a bit weird for `batch` to be a dataloader here. But I guess we can figure something out once there's a better way for handling very different `training_step` signatures.

Reply:
This is often used to process the `processed_batch_end_outputs` more than the batch. I believe it is better to call it.